azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/AvailabilitySets.java

```diff
@@ -3,25 +3,19 @@
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsDeletingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsGettingByGroup;
 import com.microsoft.azure.management.resources.fluentcore.arm.collection.SupportsListingByGroup;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsCreating;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsDeleting;
-import com.microsoft.azure.management.resources.fluentcore.collection.SupportsListing;
+import com.microsoft.azure.management.resources.fluentcore.collection.ResourcesInGroup;
 
 /**
  * Entry point to availability set management API.
  */
-public interface AvailabilitySets extends
-        SupportsListing<AvailabilitySet>,
+public interface AvailabilitySets extends 
+        ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionBlank>,
         SupportsListingByGroup<AvailabilitySet>,
         SupportsGettingByGroup<AvailabilitySet>,
-        SupportsCreating<AvailabilitySet.DefinitionBlank>,
-        SupportsDeleting,
         SupportsDeletingByGroup {
-    /**
-     * Entry point to availability set management API within a specific resource group.
-     */
-    interface InGroup extends
-            SupportsListing<AvailabilitySet>,
-            SupportsCreating<AvailabilitySet.DefinitionCreatable>,
-            SupportsDeleting {}
+    
+    
+    public interface InGroup 
+        extends ResourcesInGroup<AvailabilitySet, AvailabilitySet.DefinitionAfterGroup> {
```

Comment:
If we have a new `DefinitionWithFoo` we will have to add `withFoo` in the impl file as well.
And I'm not talking about how much work it saves us (and this is also opinionated), first it breaks the pattern of the current interface containing information about the next stage - so far in every our interface `DefinitionWithFoo` we can tell either from the interface signature or method return type what the next step is. Second it really adds a interface which really doesn't help the users at all. I'd rather spend extra 2 minutes to modify the impl every time there is a such a required parameter being added (what are the odds btw) than to have this duplicate interface appear everywhere. But sure this is also opinionated.